libs/pages/application/src/lib/feature/page-settings-domains-feature/page-settings-domains-feature.tsx

```diff
@@ -83,7 +86,15 @@ export function PageSettingsDomainsFeature() {
               dispatch(deleteCustomDomain({ applicationId, customDomain, serviceType: getServiceType(application) }))
                 .unwrap()
                 .then(() => {
-                  dispatch(fetchApplicationLinks({ applicationId }))
+                  queryClient.invalidateQueries(
+                    queries.services.listLinks({
+                      serviceId: applicationId,
+                      serviceType: getServiceType(application as ApplicationEntity) as Extract<
+                        ServiceType,
+                        'APPLICATION' | 'CONTAINER' | 'JOB' | 'CRON_JOB' | 'LIFECYCLE_JOB'
+                      >,
```

Comment:
```suggestion
                      // NOTE: we don't care about the serviceType here because it's not related to cache
                      serviceType: 'APPLICATION',
```